shell/browser/api/atom_api_web_request_ns.h

```diff
@@ -5,27 +5,59 @@
 #ifndef SHELL_BROWSER_API_ATOM_API_WEB_REQUEST_NS_H_
 #define SHELL_BROWSER_API_ATOM_API_WEB_REQUEST_NS_H_
 
+#include "base/values.h"
+#include "gin/arguments.h"
+#include "gin/handle.h"
+#include "gin/wrappable.h"
 #include "native_mate/dictionary.h"
 #include "native_mate/handle.h"
-#include "shell/browser/api/trackable_object.h"
 
 namespace electron {
 
 class AtomBrowserContext;
 
 namespace api {
 
-class WebRequestNS : public mate::TrackableObject<WebRequestNS> {
+class WebRequestNS : public gin::Wrappable<WebRequestNS> {
```

Comment:
will `gin::Wrappable` lifetime be made compatible with `mate::TrackableObject` ? Do we plan to eventually remove the `TrackableObject` ?

Reply:
For `WebRequest`, it does not really need to be a `TrackableObject` as it is managed by `Session` and it does not hold native resources.

Previously it held a reference to `BrowserContext` and thus had to be a `TrackableObject`, I think it was a design error.

`TrackableObject` is still needed for cases that need to clean up resources on shutdown.